tests/test_http_transport.py

```diff
@@ -0,0 +1,290 @@
+"""Tests for HTTP transport functionality."""
+
+import threading
+import time
+from unittest.mock import MagicMock, patch
+
+import pytest
+import requests
+
+from mcp_jupyter.server import create_server
+
+
+class TestHTTPTransport:
+    """Test HTTP transport functionality."""
+
+    def test_create_server_with_default_settings(self):
+        """Test server creation with default settings."""
+        server = create_server()
+        assert server is not None
+        assert server.name == "notebook"
+
+    def test_create_server_with_custom_host_port(self):
+        """Test server creation with custom host and port."""
+        server = create_server(host="0.0.0.0", port=9090)
+        assert server is not None
+        assert server.name == "notebook"
+
+    def test_create_server_with_stateless_mode(self):
+        """Test server creation with stateless HTTP mode."""
+        server = create_server(stateless_http=True)
+        assert server is not None
+        assert server.name == "notebook"
+
+    @pytest.mark.integration
+    def test_http_server_startup(self):
+        """Test that HTTP server can start and respond to requests."""
+        server = create_server(port=8081)
+
+        # Start server in a separate thread
+        server_thread = threading.Thread(
+            target=lambda: server.run(transport="streamable-http"), daemon=True
+        )
+        server_thread.start()
+
+        # Give server time to start
+        time.sleep(2)
```

Comment:
Instead of this, can we create some while loop helper function with a timeout that tries to connect? I think that would probably make these tests more robust